src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -11,4 +25,59 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
 		model.addAttribute("name", name);
 		return "greeting";
 	}
+
+	@Autowired
+	private BookRepository bookRepository;
+
+	@GetMapping
+	public Iterable findAll() {
+		return bookRepository.findAll();
+	}
+
+	@GetMapping("/title/{bookTitle}")
+	public List<Book> findByTitle(@PathVariable String bookTitle) {
+		return bookRepository.findBookByTitle(bookTitle);
+	}
+
+	@GetMapping("/{id}")
+	public Book findOne(@PathVariable Long id) {
+		try {
```

Comment:
If you use CrudRepository and return Optionals instead of Books, you could do something simpler like `return bookRepository.findBookById(id).orElseThrow(...);` 😃